src/test/java/com/google/edith/QueryItemsTest.java

```diff
@@ -0,0 +1,123 @@
+import org.mockito.MockitoAnnotations;
+
+public class QueryItemsTest {
+
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final QueryItems query = new QueryItems(datastore, userService);
+
+  private Map<String, Object> map =
+      ImmutableMap.of("com.google.appengine.api.users.UserService.user_id_key", "12345");
+
+  private LocalServiceTestHelper testHelper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvAttributes(map)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail")
+          .setEnvIsAdmin(true)
+          .setEnvEmail("user@gmail.com");
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    testHelper.setUp();
+    createAndStoreEntites();
+  }
+
+  @After
+  public void tearDown() {
+    testHelper.tearDown();
+  }
+
+  @Test
+  public void findExpiredItems_expiredItem_displaysExpiredItem() {
+    Assert.assertTrue(query.findExpiredItems().contains("Apple Juice"));
+  }
+
+  @Test
+  public void findExpiredItems_notExpiredItem_doesntDisplayExpiredItem() {
+    Assert.assertTrue(!query.findExpiredItems().contains("Bread"));
+  }
+
+  @Test
+  public void findExpiredItems_expiredItemInWeeks_displaysExpiredItem() {
+    Assert.assertTrue(query.findExpiredItems().contains("Peanut Butter"));
+  }
+
+  @Test
+  public void findExpiredItems_duplicateItemsInReceipt_onlyDisplaysOne() {
+    String json = query.findExpiredItems();
+    JsonParser parser = new JsonParser();
+    JsonArray inputJson = parser.parse(json).getAsJsonArray();
+    String jsonString = inputJson.toString();
+    int index = jsonString.indexOf("Peanut Butter");
+    jsonString = jsonString.substring(index + 13, jsonString.length());
+    Assert.assertTrue(jsonString.indexOf("Peanut Butter") == -1);
+  }
+
+  // Creates and Stores Receipt and Item entities in Datastore.
+  private void createAndStoreEntites() {
+    Entity receipt1 =
+        createReceiptEntity("12345", "Whole Foods", "2020-07-27", "weekend", "url1", 2.5f);
+    Entity receipt2 =
+        createReceiptEntity("12345", "Whole Foods", "2020-07-23", "sunday", "url2", 3.5f);
+    Entity receipt3 =
+        createReceiptEntity("12345", "Whole Foods", "2020-08-01", "sunday", "url2", 3.5f);
+    createItemEntity(receipt1, "12345", "Apple Juice", 5, 5.6f, "fruit", "6.0 Days");
+    createItemEntity(receipt1, "12345", "Peanut Butter", 6, 4.8f, "protein", "1.0 Weeks");
+    createItemEntity(receipt2, "12345", "Peanut Butter", 5, 2.5f, "protein", "1.0 Weeks");
+    createItemEntity(receipt3, "12345", "Bread", 3, 4.6f, "bread", "2.0 Weeks");
+  }
+
+  // Creates a Receipt Entity and stores it in Datastore.
+  private Entity createReceiptEntity(
+      String userId, String storeName, String date, String name, String fileUrl, float price) {
```

Comment:
Methods with 3+ arguments are generally discouraged, especially native type arguments. A fix would be to use a container class for these values (e.g. AutoValue) with a Builder that has named setters.

```java
createReceiptEntity(Receipt.newBuilder().setUserId(...).setStoreName(...) [...] .build());
```